omas/omas_plot.py

```diff
@@ -924,6 +1009,312 @@ def core_profiles_pressures(ods, time_index=None, ax=None, **kw):
+    Plot waves beams in poloidal cross-section
+
+    :param ods: input ods
+
+    :param time_index: time slice to plot
+
+    :param ax: axes to plot in (active axes is generated if `ax is None`)
+
+    :param kw: arguments passed to matplotlib plot statements
+
+    :return: axes handler
+    """
+
+    if time_index is None:
+        time_index = numpy.arange(len(ods['pellets'].time()))
+    if isinstance(time_index, (list, numpy.ndarray)):
+        time = ods['pellets'].time()
+        if len(time) == 1:
+            time_index = time_index[0]
+        else:
+            return ods_time_plot(pellets_trajectory_CX, time, ods, time_index, ax=ax, **kw)
+
+    import matplotlib
+    from matplotlib import pyplot
+
+    if ax is None:
+        ax = pyplot.gca()
+
+    pellets = ods['pellets']['time_slice'][time_index]['pellet']
+    for pellet in pellets:
+        R0 = pellets[pellet]['path_geometry.first_point.r']
+        R1 = pellets[pellet]['path_geometry.second_point.r']
+        Z0 = pellets[pellet]['path_geometry.first_point.z']
+        Z1 = pellets[pellet]['path_geometry.second_point.z']
+        ax.plot([R0, R1], [Z0, Z1], '--', **kw)
+
+    return ax
+
+@add_to__ODS__
+def pellets_trajectory_CX_topview(ods, time_index=None, ax=None, **kw):
+    """
+    Plot  pellet trajectory in toroidal cross-section
+
+    :param ods: input ods
+
+    :param time_index: time slice to plot
+
+    :param ax: axes to plot in (active axes is generated if `ax is None`)
+
+    :param kw: arguments passed to matplotlib plot statements
+
+    :return: axes handler
+    """
+    if time_index is None:
+        time_index = numpy.arange(len(ods['pellets'].time()))
+    if isinstance(time_index, (list, numpy.ndarray)):
+        time = ods['pellets'].time()
+        if len(time) == 1:
+            time_index = time_index[0]
+        else:
+            return ods_time_plot(pellets_trajectory_CX_topview, time, ods, time_index, ax=ax, **kw)
+
+    import matplotlib
+    from matplotlib import pyplot
+
+    if ax is None:
+        ax = pyplot.gca()
+
+    pellets = ods['pellets']['time_slice'][time_index]['pellet']
+    for pellet in pellets:
+        R0 = pellets[pellet]['path_geometry.first_point.r']
+        R1 = pellets[pellet]['path_geometry.second_point.r']
+        phi0 = pellets[pellet]['path_geometry.first_point.phi']
+        phi1 = pellets[pellet]['path_geometry.second_point.phi']
+
+        x0 = R0 * numpy.cos(phi0)
+        y0 = R0 * numpy.sin(phi0)
+
+        x1 = R1 * numpy.cos(phi1)
+        y1 = R1 * numpy.sin(phi1)
+        ax.plot([x0, x1], [y0, y1], '--', **kw)
+
+    return ax
+
+@add_to__ODS__
+def lh_antennas_CX(ods, time_index=0, ax=None, antenna_plotlength=None, **kw):
+    """
+    Plot LH antenna position in poloidal cross-section
+
+    :param ods: input ods
+
+    :param time_index: time slice to plot
+
+    :param ax: axes to plot in (active axes is generated if `ax is None`)
+
+    :param antenna_plotlength: length of antenna on plot
```

Comment:
I personally find the `antenna_plotlength` and `launcher_plotlength` not descriptive of what this variable sets. I think it would call it something like `antenna_trajectory` and `launcher_trajectory` and use the matplotlib `arrow` method instead.

Reply:
I am currently using markers to differentiate between different actuators. I don't see anyway to do something similar with the arrow method

Reply:
Maybe you could overplot the marker and the arrow...